### DISCUSSION TITLE
Inquiry about the difference of GND density map

### DISCUSSION THREAD

**Discussion by jdkang0810** [mtex-toolbox/mtex#2258] [Ask Anything] [answered]
Thank you for providing the MTEX platform for analyzing EBSD data. I appreciate the tools and support your team offers.
I am currently working on calculating and analyzing Geometrically Necessary Dislocation (GND) density maps using MTEX. 
However, I have encountered a discrepancy between the GND density maps generated by MTEX and those produced by Aztecrystal.

First figure is produced using Azetecrystals.
![0 5_110_2](https://github.com/user-attachments/assets/7b72aab6-3bbd-4ccd-ab84-710a5ef0afce)
High GND density is indicated by the green area. 

Here are the specifics of my setup:

Space Group: Fm-3m
Slip Direction: [1, -1, 0]
Slip Plane: {1, 1, 1}
Burgers Vector: 1/2[1, -1, 0]

I used the following approach in my code:
1. Configured the space group, slip direction, slip plane, and Burgers vector as mentioned above.
2. Utilized the gridify function to obtain the EBSDsquare class.
3. Calculated the GND density based on the generated EBSDsquare.

```
 %% Import Script for EBSD Data
 img_CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [2.8 2.8 2.8], 'mineral', 'Fe', 'color', [0.53 0.81 0.98])};
 
 % plotting convention
 setMTEXpref('xAxisDirection','east');   setMTEXpref('zAxisDirection','outOfPlane');
 
 % Specify File Names
 pname = 'C:\Temp\';
 img_fname = [pname '\tester
[tester.zip](https://github.com/user-attachments/files/17718379/tester.zip)
.ctf'];
 
 img_ebsd_all = EBSD.load(img_fname,img_CS,'interface','ctf','convertSpatial2EulerReferenceFrame'); 
 
 img_targetPhaseName = 'Fe';                       
 ebsd = img_ebsd_all(img_targetPhaseName);         
 CS = ebsd.CS;     
 
 %% 1. calc grains
 [grains,ebsd.grainId] = calcGrains(ebsd,'angle',2*degree,'minPixel',10);
 figure(11);     plot(grains,grains.boundary,'linewidth',1)
 figure(14);     plot(ebsd,grains.boundary,'linewidth',1)
 
 % define the color key
 ipfKey = ipfHSVKey(ebsd);
 ipfKey.inversePoleFigureDirection = yvector;
 
 % smooth grain boundaries
 grains = smooth(grains,5);
 
 figure(12)
 plot(ebsd,ipfKey.orientation2color(ebsd.orientations),'micronBar','off','figSize','medium')
 hold on ;   plot(grains.boundary,'linewidth',1);    hold off;
 
 ipfKey = axisAngleColorKey(ebsd);                       
 ipfKey.oriRef = grains(ebsd.grainId).meanOrientation;   
 
 % plot the data
 figure(13)
 plot(ebsd,ipfKey.orientation2color(ebsd.orientations),'micronBar','off','figSize','medium')
 hold on;    plot(grains.boundary,'linewidth',1);    hold off;
 
 %% define the denoising filter
 F = halfQuadraticFilter;
 
 ebsd2 = smooth(ebsd,F,'fill',grains);
 ebsd2 = ebsd2('Fe');
 [grains2, ebsd2.grainId] = calcGrains(ebsd2, 'angle', 1*degree, 'minPixel', 10);
 
 grains2 = smooth(grains2, 5);
 
 ipfKey.oriRef = grains2(ebsd2.grainId).meanOrientation;
 
 figure(15);
 plot(ebsd2, ipfKey.orientation2color(ebsd2.orientations), 'micronBar','off','figSize','medium');
 hold on;    plot(grains2.boundary, 'linewidth', 1); hold off;
 
 %%
 ebsdGrid = gridify(ebsd2);
 % ebsdGrid = fill(ebsdGrid);
 kappa = ebsdGrid.curvature;
 newMtexFigure('nrows',3,'ncols',3);
 
 for i = 1:3
   for j = 1:3
     nextAxis(i,j)
     plot(ebsdGrid,kappa{i,j},'micronBar','off')
     hold on; plot(grains2.boundary,'linewidth',1); hold off
   end
 end
 
 setColorRange([-0.4, 0.4])
 drawNow(gcm,'figSize','large')
 
 %%
 slipPlane = Miller(1, 1, 1, ebsdGrid.CS); 
 slipDirection = Miller(1, -1, 0, ebsdGrid.CS); 
 
 b_magnitude = 1/2 * norm(slipDirection);
 
dislocationSystem_cubic = dislocationSystem(slipPlane, slipDirection);
 [gndDensity, gndTensor] = calcGND(ebsdGrid, dislocationSystem_cubic);
 
 ebsdGrid.prop.gndDensity = gndDensity;
 %% Plot the GND density map
 figure(21);
 plot(ebsdGrid, ebsdGrid.prop.gndDensity, 'micronbar', 'off');
 clim([0 5e17]); 
 mtexColorbar('title', '\rho_{GND} [m^{-2}]');
```
The resultant figure by MTEX is so different to figure by Azetecrystal. 
![image](https://github.com/user-attachments/assets/70ab4ab3-b2e2-40cb-a24f-bca02dfc6432)

The results I obtained from MTEX show GND density values reaching approximately 5*10^17, whereas Aztecrystal produces values around 10^16. 
This significant difference has in magnitude and shape. I'm not sure which one is correct or if I did something right.

Could you please help me understand the possible reasons for this discrepancy?
Additionally, I would appreciate any guidance or solutions you might suggest to align the GND density results between MTEX and Aztecrystal.

Thank you for your assistance and support.

**Comment by kilir** [comment, marked as answer]
Hi,

one main point you are missing is that you do not symmetrise your dislocation system. Hence, your solution will need more dislocations in an attempt to fit the obsered gradient. 

Try
```Matlab
dislocationSystem_cubic = symmetrise(dislocationSystem(slipPlane, slipDirection))
```

Further differences will arise from the way you conduct the noise removal, grain reconstruction etc... and potentially the way the GND are computed (I somewhat doubt that aztec does it exactly the way mtex does).

Hope that helps.

Cheers,

Rüdiger

**Reply by jdkang0810** [reply]
Dear Rüdiger,

Thank you for your kind explaination.
I can produce similar figure following your instruction.
![result](https://github.com/user-attachments/assets/4a487340-e25a-4cb0-bb4b-8f1280e06617)
